src/table.rs

```diff
@@ -316,21 +320,30 @@ impl Table {
 
     fn update_current_player_index_to_next_active(&mut self) {
         for _ in 0..self.players.len() {
-            // Set the next active player to the next index, looping if too large
+            // Set the next active player to the next index, resetting back down if out of bounds
             self.current_player_index += 1;
             if self.current_player_index >= self.players.len() {
-                self.current_player_index = 0;
+                self.current_player_index -= self.players.len();
```

Comment:
What case does this change cover?

Reply:
That was just some insanity left over before I understood the issue. It can be reverted